policyengine_taxsim/core/input_mapper.py

```diff
@@ -1,10 +1,95 @@
 from .utils import (
     load_variable_mappings,
-    get_state_code,
+    get_state_code, get_ordinal,
 )
+import copy
 
 
-def import_single_household(taxsim_vars):
+def add_additional_tax_units(state, year, situation):
+    has_use_tax = ['pa', 'nc', 'ca', 'il', 'in', 'ok']
+    if state in has_use_tax:
+        situation["tax_units"]["your tax unit"][f"{state}_use_tax"] = {str(year): 0.0}
+    return situation
+
+
+def form_household_situation(year, state, taxsim_vars):
+    mappings = load_variable_mappings()["taxsim_to_policyengine"]
+
+    household_situation = copy.deepcopy(mappings["household_situation"])
+
+    depx = taxsim_vars["depx"]
+    mstat = taxsim_vars["mstat"]
+
+    if mstat == 2:  # Married filing jointly
+        members = ["you", "your partner"]
+    else:  # Single, separate, or dependent taxpayer
+        members = ["you"]
+
+    for i in range(1, depx + 1):
+        members.append(f"your {get_ordinal(i)} dependent")
+
+    household_situation["families"]["your family"]["members"] = members
+    household_situation["households"]["your household"]["members"] = members
+    household_situation["tax_units"]["your tax unit"]["members"] = members
+
+    household_situation = add_additional_tax_units(state.lower(), year, household_situation)
+
+    household_situation["spm_units"]["your household"]["members"] = members
+
+    if depx > 0:
+        household_situation["marital_units"] = {
+            "your marital unit": {
+                "members": ["you", "your partner"] if mstat == 2 else ["you"]
+            }
+        }
+        for i in range(1, depx + 1):
+            dep_name = f"your {get_ordinal(i)} dependent"
+            household_situation["marital_units"][f"{dep_name}'s marital unit"] = {
+                "members": [dep_name],
+                "marital_unit_id": {str(year): i}
+            }
+    else:
+        household_situation["marital_units"]["your marital unit"]["members"] = (
+            ["you", "your partner"] if mstat == 2 else ["you"]
+        )
```

Comment:
We are just adding the dependents to the marital unit right? 
I think we should be adding them as members to the tax_unit, household, family, and spm_unit as well

Reply:
After looking into the issue, it is passed through to all entities